crates/usvg/src/tree/mod.rs

```diff
@@ -628,12 +631,29 @@ impl Default for FillRule {
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub(crate) enum ContextElement {
+    /// The current context element is a use node. Since we can get
+    /// the bounding box of a use node only once we have converted
+    /// all elements, we need to fix the transform and units of
+    /// the stroke/fill after converting the whole tree.
+    UseNode,
+    /// The current context element is a path node (i.e. only applicable
+    /// if we draw the marker of a path). Since we already know the bounding
+    /// box of the path when rendering the markers, we can convert them directly,
+    /// so we do it while parsing.
+    PathNode(Transform, NonZeroRect),
+}
+
 /// A fill style.
 #[derive(Clone, Debug)]
 pub struct Fill {
     pub(crate) paint: Paint,
     pub(crate) opacity: Opacity,
     pub(crate) rule: FillRule,
+    // Whether the current fill needs to be resolved relative
+    // to a context element.
+    pub(crate) context_element: Option<ContextElement>,
```

Comment:
The only reason we have to store it is because of paint bbox resolver?
I think we can avoid it for markers, since we already know the bbox beforehand, unlike with text.
But I guess I will try figuring it out by myself, since it would be a bit tricky.

Reply:
Yes, we mainly need it as a "marker" so that when iterating over all paint servers we know which strokes/fills need to be converted.